src/lib-imap/imap-capability-list.h

```diff
@@ -0,0 +1,85 @@
+#ifndef IMAP_CAPABILITY_LIST_H
+#define IMAP_CAPABILITY_LIST_H
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include "lib.h"
+#include "array.h"
+
+/* bitflags to represent different conditions of
+   the current imap session */
+enum imap_capability_visibility {
+	IMAP_CAP_VISIBILITY_NEVER           = 0,      /* never show this capability */
+	IMAP_CAP_VISIBILITY_PREAUTH         = BIT(0), /* show this cap before login */
+	IMAP_CAP_VISIBILITY_POSTAUTH        = BIT(1), /* show this cap after login */
+	IMAP_CAP_VISIBILITY_TLS_ACTIVE      = BIT(2), /* show this cap if tls is active */
+	IMAP_CAP_VISIBILITY_TLS_INACTIVE    = BIT(3), /* show this cap if tls is inactive */
+	IMAP_CAP_VISIBILITY_NO_LOGIN        = BIT(6), /* show this cap if login is disabled */
+	/* NOTE: connections from localhost are 'secure' regardless of SSL/TLS! */
+	IMAP_CAP_VISIBILITY_SECURE          = BIT(7), /* show this cap if we are secured */
+	IMAP_CAP_VISIBILITY_INSECURE        = BIT(8), /* show this cap if we aren't secured */
+
+	/* IMAP_CAP_VISIBILITY_REQUIRE_ALL indicates that the rest
+	   of the conditions must be met *exactly* to show this
+	   capability, otherwise if *at least* 1 condition is met
+	   the capability will be shown */
+	IMAP_CAP_VISIBILITY_REQUIRE_ALL     = BIT(31),
+
+	IMAP_CAP_VISIBILITY_ALWAYS          = 0x7FFFFFFF /* always show this cap */
+};
+
+/*  the initial size of the capability list
+	it gets automatically increased but this
+	is the size that we start at */
+#define INITIAL_CAP_LIST_SIZE 5
```

Comment:
This is not used anywhere. So this is the one I suggested to be 64 earlier. But it's not really needed to be a macro, since it's clear enough anyway. -> remove